File: ruby-gem/lib/models/as_realized_transformation.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #AsRealizedTransformation
require_relative '../core/semantic_object'
require_relative 'transformation'


module DfcLinkmlConnector
  module Models
    class AsRealizedTransformation < Transformation
      SEMANTIC_TYPE = "dfc-b:AsRealizedTransformation".freeze

      # @return [Float]
      attr_accessor :cost

      # @return [String]
      attr_accessor :end_date

      # @return [String]
      attr_accessor :start_date

      # @return [String]
      attr_accessor :input

      # @return [String]
      attr_accessor :output

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, cost: nil, endDate: nil, startDate: nil, input: nil, output: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, cost: nil, endDate: nil, startDate: nil, input: nil, output: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @cost = cost
        @end_date = endDate
        @start_date = startDate
        @input = input
        @output = output
        self.semanticType = "dfc-b:AsRealizedTransformation"
        registerSemanticProperty("dfc-b:cost", &method("cost")).valueSetter = method("cost=")
        registerSemanticProperty("dfc-b:endDate", &method("end_date")).valueSetter = method("end_date=")
        registerSemanticProperty("dfc-b:startDate", &method("start_date")).valueSetter = method("start_date=")
        registerSemanticProperty("dfc-b:hasInput", &method("input")).valueSetter = method("input=")
        registerSemanticProperty("dfc-b:hasOutput", &method("output")).valueSetter = method("output=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end

File: typescript-connector/src/models/AsRealizedTransformation.ts
// Class from DFC Business Ontology: #AsRealizedTransformation
import { SemanticObject } from "../core/SemanticObject.js";
import { Transformation, type TransformationParams } from "./Transformation.js";

export interface AsRealizedTransformationParams extends TransformationParams {
  cost?: number;
  endDate?: string;
  startDate?: string;
  hasInput?: string;
  hasOutput?: string;
}
export class AsRealizedTransformation extends Transformation {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:AsRealizedTransformation";
  }

  cost?: number;
  endDate?: string;
  startDate?: string;
  hasInput?: string;
  hasOutput?: string;

  constructor(
    semanticId: string,
    params?: AsRealizedTransformationParams,
  ) {
    super(semanticId, params);
    this.cost = params?.cost;
    this.endDate = params?.endDate;
    this.startDate = params?.startDate;
    this.hasInput = params?.hasInput;
    this.hasOutput = params?.hasOutput;
    this.semanticType = AsRealizedTransformation.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:cost", () => this.cost);
    this.registerSemanticProperty("dfc-b:endDate", () => this.endDate);
    this.registerSemanticProperty("dfc-b:startDate", () => this.startDate);
    this.registerSemanticProperty("dfc-b:hasInput", () => this.hasInput);
    this.registerSemanticProperty("dfc-b:hasOutput", () => this.hasOutput);
  }
  static {
    SemanticObject.typeRegistry.set(AsRealizedTransformation.SEMANTIC_TYPE, AsRealizedTransformation);
  }
}
